File: app/src/main/java/com/alexharman/stitchathon/KnitPackage/Stitch.java
package com.alexharman.stitchathon.KnitPackage;

public class Stitch {
    private int width;
    private String type;
    public boolean done;

    Stitch(String type, int width) {
        this.width = width;
        this.type = type;
    }

    Stitch(String type) {
        this(type, 1);
    }

    public int getWidth() {
        return width;
    }

    public String getType() {
        return type;
    }
}

File: app/src/main/java/com/alexharman/stitchathon/KnitPackage/Stitch.kt
package com.alexharman.stitchathon.KnitPackage

data class Stitch constructor(val type: String, val width: Int = 1) {
    val isSplit: Boolean = type.contains("/")
    val madeOf: Array<Stitch> =
            if (isSplit)
                type.split("/").map { Stitch(it) }.toTypedArray()
            else
                arrayOf()

    override fun toString(): String = type
}
